### PR TITLE
Refactor domain initialization

### LINKED ISSUE #2123: Domain Queries for CurrentUser and GeneralSettings
- currently located at `assets\prototype\domain\eventEditor\data\queries\currentUser` and `assets\prototype\domain\eventEditor\data\queries\generalSettings` but are not exclusive to the Event Editor
- create `assets\prototype\domain\shared\data\queries\`
- move the above two folders into new location
- add tests for:
    - `useCurrentUser`
    - `useFetchCurrentUser`
    - `useFetchGeneralSettings`
    - `useGeneralSettings`

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Moves initial queries and cache rehydration to `/domain/eventEditor/data/initialization`
- Separates/Creates cache updaters inside `/domain/shared/data/queries` to make them reusable
- Moves `currentUser` and `generalSettings` queries to `/domain/shared/data/queries`
- Simplifies Cache Rehydration
- Closes #2123, #2122

### CODE REVIEW

File: assets/prototype/application/services/context/ConfigProvider.tsx [hunk 1]
```diff
@@ -5,8 +5,7 @@ import React, { createContext, useState } from 'react';
 import { useConfigData, ConfigDataProps, Config } from '../config';
 import { CurrentUser, DateTimeFormats } from '../../valueObjects/config';
 
-import useCurrentUser from '../../../domain/eventEditor/data/queries/currentUser/useCurrentUser';
-import useGeneralSettings from '../../../domain/eventEditor/data/queries/generalSettings/useGeneralSettings';
+import { useCurrentUser, useGeneralSettings } from '../../../domain/shared/data/queries';
```
Comment (nicolad) [inline assets/prototype/application/services/context/ConfigProvider.tsx]: Love this result :+1:

File: assets/prototype/domain/shared/data/queries/currentUser/useFetchCurrentUser.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { FetchEntitiesResult } from '../../../../eventEditor/data/queries/types';
+import { Viewer } from '../../../../../application/valueObjects/config/types';
+
+const useFetchCurrentUser = (): FetchEntitiesResult<Viewer> => {
+	const { data, error, loading } = useQuery<Viewer>(GET_CURRENT_USER);
+
+	return {
+		data,
+		error,
+		loading,
```
Comment (nicolad) [inline assets/prototype/domain/shared/data/queries/currentUser/useFetchCurrentUser.ts]: Should these handled by the toaster, or that would happen on a different step?

Comment (manzoorwanijk) [reply assets/prototype/domain/shared/data/queries/currentUser/useFetchCurrentUser.ts]: Any consumer can make use of that but we haven't added that to toaster yet.

Comment (manzoorwanijk) [reply assets/prototype/domain/shared/data/queries/currentUser/useFetchCurrentUser.ts]: Any consumer can make use of that but we haven't added that to toaster yet.

File: assets/prototype/domain/shared/data/queries/entities/useUpdateDatetimeList.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { DatetimesList } from '../../../../eventEditor/data/types';
+import { WriteQueryOptions } from '../../../../eventEditor/data/queries/types';
+import useUpdateEntityList from './useUpdateEntityList';
+import useDatetimeQueryOptions from '../../../../eventEditor/data/queries/datetimes/useDatetimeQueryOptions';
+import { CacheUpdaterFn } from '../types';
```
Comment (tn3rb) [inline assets/prototype/domain/shared/data/queries/entities/useUpdateDatetimeList.ts]: this shared logic seems to be pulling in a lot of stuff from the `eventEditor` domain...
can that other logic be shared too? or is this logic maybe still too coupled to the event editor?
maybe this shouldn't be shared if it is tied to the event editor this much?

**Comment by tn3rb** [comment]
I'm thinking for folder structure, we might want to add some additional levels to the domain folder like:

```
├── assets
|	├── src
|	|	├── application
|	|	|	├── config
|	|	|	├── services
|	|	|	├── ui
|	|	├── domain
|	|	|	├── admin
|	|	|	|	├── events
|	|	|	|	|	├── editor
|	|	|	|	|	├── listTable
|	|	|	|	|	├── settings
|	|	|	|	├── transactions
|	|	|	|	|	├── details
|	|	|	|	|	├── listTable
|	|	|	|	|	├── settings
|	|	|	|	├── registrations
|	|	|	|	|	├── details
|	|	|	|	|	├── listTable
|	|	|	|	|	├── settings
|	|	|	├── public
|	|	|	|	├── events
|	|	|	|	|	├── archive
|	|	|	|	|	├── calendar
|	|	|	|	|	├── single
|	|	|	├── shared
|	|	├── infrastructure
|	|	|	├── services
|	|	|	|	├── apollo
```

File: assets/prototype/domain/eventEditor/data/initialization/useCacheRehydration.ts [hunk 4]
```diff
@@ -36,83 +42,76 @@ const useCacheRehydration = (): void => {
 	} = useCacheRehydrationData();
 	const { isLoaded } = useStatus();
 
+	const datetimeIn = pathOr<Datetime[]>([], ['nodes'], espressoDatetimes).map(({ id }) => id);
+	const ticketIn = pathOr<Ticket[]>([], ['nodes'], espressoTickets).map(({ id }) => id);
```
[the remark this answers is not in the document]
Comment (nicolad) [reply assets/prototype/domain/eventEditor/data/initialization/useCacheRehydration.ts]: I was thinking to suggest ramda's `pipe`, but thought it's overkill for 2 steps;
+1 for `getEntityId` or similar

File: assets/prototype/domain/eventEditor/data/queries/prices/usePriceQueryOptions.ts [hunk 5]
```diff
@@ -3,13 +3,13 @@ import { GET_PRICES } from './';
...
-const usePriceQueryOptions = (): ReadQueryOptions => {
-	const ticketIn: EntityId[] = useTicketIds();
+const usePriceQueryOptions = (ticketIn: EntityId[] = []): ReadQueryOptions => {
+	const ticketIds = useTicketIds();
 	const options: ReadQueryOptions = {
 		query: GET_PRICES,
 		variables: {
 			where: {
-				ticketIn,
+				ticketIn: ticketIn.length ? ticketIn : ticketIds,
```
[the remark this answers is not in the document]
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/data/queries/prices/usePriceQueryOptions.ts]: Remember that bug that triggered console warning? We can't use hooks inside conditionals.

File: assets/prototype/domain/shared/data/queries/entities/useUpdateDatetimeList.ts (hunk 3 above)
Comment (manzoorwanijk) [reply assets/prototype/domain/shared/data/queries/entities/useUpdateDatetimeList.ts]: Yes, didn't notice that. Thank you :)

File: assets/prototype/domain/eventEditor/data/initialization/useCacheRehydration.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/data/initialization/useCacheRehydration.ts]: Agreed 👍

File: assets/prototype/domain/eventEditor/data/queries/prices/usePriceQueryOptions.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (nicolad) [reply assets/prototype/domain/eventEditor/data/queries/prices/usePriceQueryOptions.ts]: I've used `skip` for `useQuery`, not sure if that option is also available for `ReadQueryOptions`.

**Comment by manzoorwanijk** [comment]
@tn3rb Since those cache updaters that I created in `/domain/shared/data/queries/entities` update the cache that's specific to `domain/eventEditor` Apollo client, it makes more sense to move them into `domain/eventEditor` instead of `domain/shared`.

File: assets/prototype/domain/eventEditor/data/queries/prices/usePriceQueryOptions.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/data/queries/prices/usePriceQueryOptions.ts]: No, that's not available

File: assets/prototype/domain/eventEditor/data/initialization/useCacheRehydration.ts [hunk 6]
```diff
@@ -23,7 +23,7 @@ import { useUpdateCurrentUserCache, useUpdateGeneralSettingsCache } from '../../
 const { GET_CURRENT_USER, GET_GENERAL_SETTINGS } = queries;
 
 const useCacheRehydration = (): void => {
-	const { setData } = useRelations();
+	const { setData: setRelationalData } = useRelations();
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/initialization/useCacheRehydration.ts]: Great :+1:

**Pull request merged**